### DISCUSSION TITLE
T-Display-S3 AMOLED - animated GIFs not displaying correctly

### DISCUSSION THREAD

**Discussion by 3Dsf-info** [moononournation/Arduino_GFX#310] [Q&A]
Hi! Thanks for implementing support for the T-Display-S3 AMOLED board - much appreciated!

I've been able to upload an animated GIF file using SPIFFS and have compiled the new Arduino_GFX code. Unfortunately, the different GIFs I've tried don't display correctly. The colour mapping appears to be incorrect. It's not simply inverted. I've tried using the supplied Earth rotation animation as well.

Is this something that could be looked at? Or is there a setting I need to change? Thank you!

![gif](https://github.com/moononournation/Arduino_GFX/assets/131314273/fbe0aa55-04f5-4165-b3ba-a6ed64bbc3fe)

**Comment by moononournation** [comment]
Just tested latest code did not have this problem, please get update Arduino_GFX

**Comment by 3Dsf-info** [comment]
Thanks for the quick response!

So I removed the existing Arduino_GFX library from my machine, and downloaded the latest. (1.3.6)  I added the updated data bus and display classes to ImgViewerAnimatedGIF.ino.

Arduino_DataBus *bus = new Arduino_ESP32QSPI(
    6 /* cs */, 47 /* sck */, 18 /* d0 */, 7 /* d1 */, 48 /* d2 */, 5 /* d3 */);

Arduino_GFX *gfx = new Arduino_RM67162(bus, 17 /* RST */, 0 /* rotation */);

I commented out LittleFS and uncommented SPIFFS.

I then recompiled, using the supplied GIF file of the rotating Earth.

I'm afraid that the same thing happens - the background and Earth animation is green. This is using Arduino 1.8.19 on a Mac. I tried Arduino 2.10 and the same thing happened. I don't normally use 2.10 because they removed the ability to upload files to SPIFFs.

Am I doing something wrong? Thank you.

**Reply by moononournation** [reply]
I tested in 1.8.19, everything fine

**Reply by moononournation** [reply]
https://github.com/moononournation/Arduino_GFX/assets/13228891/e8fd79ce-a5ea-4b9e-88bc-e33a158108b9

**Comment by 3Dsf-info** [comment]
Hmmm. Would you have any suggestions where I can look? As noted, I just downloaded the current version of your library straight from Github.

**Comment by moononournation** [comment]
In case you are not, please also use latest example files.

**Comment by 3Dsf-info** [comment]
Yes, I edited the example file directly.

**Comment by 3Dsf-info** [comment]
Okay. So it's weird. I deleted all traces of Arduino_GFX from my machine. I then downloaded a full install.

I edited ImgViewerAnimatedGIF.ino, adding the new data bus and display classes. In the "Wio Terminal" section I commented out LittleFS etc, and uncommented SPIFFs.

I recompiled the code, re-uploaded the spinning Earth graphic, reset the device...

...and it displayed green again.

I wonder what I'm missing that this isn't working!

**Reply by moononournation** [reply]
I tested LittleFS ok.

**Reply by 3Dsf-info** [reply]
Maybe I'll try that. Did you use an uploader plugin with Arduino 1.8?

**Reply by moononournation** [reply]
I highly recommend you try the default setting first before you change anything.

**Reply by 3Dsf-info** [reply]
Okay. So I deleted the Arduino IDE and all associated files, and reinstalled from scratch. I went for LittleFS instead, and used default settings for everything. I couldn't upload anything - I got the error message "Error: esptool not found!" Fortunately I found the link you posted about adding an alias.
 
So now I've got the most recent version of Arduino_GFX running from scratch, with the default spinning Earth GIF stored using LittleFS.

And the graphic is still green, I'm afraid. :(

**Reply by moononournation** [reply]
Is it display correct with PDQgraphicstest example?

**Reply by 3Dsf-info** [reply]
Unfortunately, no. I compiled PDQgraphicstest and the same problem occurred - a green background.

In case it's a hardware issue I tried a second Lilygo Amoled board, and the same problem occurred.

I also tried the Volos animation demo, and it worked fine. The graphics appeared just as in his video, with no green.

https://github.com/VolosR/TDisplayAmoled

![green](https://github.com/moononournation/Arduino_GFX/assets/131314273/65468ab3-2a17-4629-a7f6-dafb58041737)

**Comment by 3Dsf-info** [comment]
Note: I'm pasting the DataBus and GFX definitions directly into the code. (I couldn't see the new Amoled definitions in the header files) That shouldn't matter, should it?

```
Arduino_DataBus *bus = new Arduino_ESP32QSPI( 6 /* cs */, 47 /* sck */, 18 /* d0 */, 7 /* d1 */, 48 /* d2 */, 5 /* d3 */);

Arduino_GFX *gfx = new Arduino_RM67162(bus, 17 /* RST */, 0 /* rotation */);

```

**Reply by moononournation** [reply]
and remove GFX_BL #define

**Comment by 3Dsf-info** [comment]
Wow! That was it! The backlight reference is all it was.

Thank you, good sir. :)

**Reply by moononournation** [reply]
AMOLED did not have backlight

**Comment by 3Dsf-info** [comment]
Of course. I had no idea that that line would make everything green, though!
